### PR TITLE
Fixes comparing content fields in Liquid

### LINKED ISSUE #15625: Liquid filters and comparison stopped working after upgrade to 1.9
### Describe the bug
Another breaking change is in our Liquid views. What stopped working:

````
{% assign my_variable_1 = "some_stuff" %}
{% assign my_variable_2 = "some_stuff" %}
{% if my_variable_1 == my_variable_2 %}
   {{ "Never here" }}
{% endif %}
````

`{% assign formated_datetime = Model.ContentItem.CreatedAtUtc | date: "%y-%M-%d" %}` this return and empty string

UNLESS a do some hacking and use the `raw` filter like so:
````
{% assign my_variable_1 = "some_stuff" | raw %}
{% assign my_variable_2 = "some_stuff" | raw %}
{% if my_variable_1 == my_variable_2 %}
   {{ "Success" }}
{% endif %}
````
`{% assign formated_datetime = Model.ContentItem.CreatedAtUtc | raw | date: "%y-%M-%d" %}`

### PR DESCRIPTION
**Pull request by gvkries**
Fixes #15625

/cc @sebastienros

### CODE REVIEW

File: test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        public async Task ComparingTextField_ReturnsCorrectValue()
+        {
+            var context = new SiteContext();
+            await context.InitializeAsync();
+            await context.UsingTenantScopeAsync(async scope =>
+            {
+                var template = """
+                    {% if Model.ContentItem.Content.MyPart.myField.Text == "Some test value" %}true{% else %}false{% endif %}
+                    """;
+
```
Comment (hishamco) [inline test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs:22]:
```suggestion
```

File: test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                Assert.Equal("true", result);
+            });
+        }
+
+        public class MyPart : ContentPart
+        {
+            public string Text { get; set; }
+        }
+
+
```
Comment (hishamco) [inline test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs:120]:
```suggestion
```

File: test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        public class MyField : ContentField
+        {
+            public int Value { get; set; }
+        }
+
```
Comment (hishamco) [inline test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs:125]:
```suggestion
```

File: test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                {
+                    x.GetOrCreate<MyField>("myField");
+                    x.Alter<MyField>("myField", f => f.Value = 123);
+                });
+
+                var json = JConvert.SerializeObject(contentItem);
+
+                var testModel = JConvert.DeserializeObject<ContentItem>(json);
+
+
```
Comment (hishamco) [inline test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs:105]:
```suggestion
```

File: test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                {
+                    x.GetOrCreate<DateTimeField>("myField");
+                    x.Alter<DateTimeField>("myField", f => f.Value = new DateTime(2024, 5, 14, 8, 0, 0, DateTimeKind.Utc));
+                });
+
+                var json = JConvert.SerializeObject(contentItem);
+
+                var testModel = JConvert.DeserializeObject<ContentItem>(json);
+
+
```
Comment (hishamco) [inline test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs:71]:
```suggestion
```

**Review by hishamco — CHANGES_REQUESTED** [review]

File: test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs (hunk 1 above)
Comment (gvkries) [reply test/OrchardCore.Tests/DisplayManagement/LiquidTests.cs:22]: Let's keep this line for better readability.

**Review by sebastienros — APPROVED** [review]

**Pull request merged**